aquadoggo/src/db/models/entry.rs

```diff
@@ -1,23 +1,23 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
-use std::convert::TryFrom;
-
-use p2panda_rs::entry::{EntrySigned, LogId, SeqNum};
-use p2panda_rs::hash::Hash;
-use p2panda_rs::identity::Author;
-use p2panda_rs::operation::OperationEncoded;
+use std::convert::{TryFrom, TryInto};
 
+use p2panda_rs::storage_provider::errors::EntryStorageError;
 use serde::Serialize;
-use sqlx::{query, query_as, FromRow};
+use sqlx::FromRow;
 
-use crate::db::Pool;
-use crate::errors::Result;
+use p2panda_rs::entry::{decode_entry, Entry as P2PandaEntry, EntrySigned, LogId, SeqNum};
+use p2panda_rs::hash::Hash;
+use p2panda_rs::identity::Author;
+use p2panda_rs::operation::{Operation, OperationEncoded};
+use p2panda_rs::storage_provider::models::EntryWithOperation;
+use p2panda_rs::storage_provider::traits::AsStorageEntry;
 
 /// Struct representing the actual SQL row of `Entry`.
 ///
 /// We store the u64 integer values of `log_id` and `seq_num` as strings since not all database
 /// backend support large numbers.
-#[derive(FromRow, Debug, Serialize)]
+#[derive(FromRow, Debug, Serialize, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct EntryRow {
```

Comment:
I don't think we need both `Entry` and `EntryRow` structs in this crate anymore as we can use types (probably `EntryWithOperation`) from `p2panda-rs`. Haven't quite figured out if this is true though.... any thoughts?